### ISSUE TITLE
Cannot start a heterogeneous image

### ISSUE THREAD

**Issue by kometchtech** [docker/docker#25250]
**Output of `docker version`:**

```
$ docker version 
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:11:10 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:11:10 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
$ docker version 
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:11:10 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:11:10 2016
 OS/Arch:      linux/amd64
kometch@ubuntu-virt:~$ docker info 
Containers: 4
 Running: 0
 Paused: 0
 Stopped: 4
Images: 20
Server Version: 1.12.0
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: xfs
 Dirs: 25
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor seccomp
Kernel Version: 4.4.0-31-generic
Operating System: Ubuntu 16.04.1 LTS
OSType: linux
Architecture: x86_64
CPUs: 24
Total Memory: 70.85 GiB
Name: ubuntu-virt.kometch.local
ID: AJPJ:MUFZ:PWTZ:TKHS:VWZ5:6FQT:QTAY:TY4H:EOVX:SLSE:QBIP:V3QW
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

```
Handle 0x0002, DMI type 1, 27 bytes
System Information
        Manufacturer: Dell                  
        Product Name: CS24-TY               
        Version: A00                   
        Serial Number: 2BFQPN1               
        UUID: 44454C4C-4200-1046-8051-B2C04F504E31
        Wake-up Type: Power Switch
        SKU Number: To Be Filled By O.E.M.
        Family: Server
```

**Steps to reproduce the issue:**
1. docker pull
2. docker run
3. standard_init_linux.go:175: exec user process caused "no such file or directory"

**Describe the results you received:**
```
kometch@ubuntu-virt:~$ docker run ioft/armhf-ubuntu
Unable to find image 'ioft/armhf-ubuntu:latest' locally
latest: Pulling from ioft/armhf-ubuntu

77c45a60f4e6: Already exists 
7d1fb2204ca8: Already exists 
8789c51a4fa7: Already exists 
Digest: sha256:18046759911b86af563d1eb8518d1dc25b873a4c62b8aea2d185e6052e4adf92
Status: Downloaded newer image for ioft/armhf-ubuntu:latest
standard_init_linux.go:175: exec user process caused "no such file or directory"
kometch@ubuntu-virt:~$ docker start 0b32ff22d617
0b32ff22d617
kometch@ubuntu-virt:~$ docker exec -it 0b32ff22d617 bash
Error response from daemon: Container 0b32ff22d6171b1109ef8f3223750513808a6e52f8c1cb3ba356cafae44b6823 is not running
```

**Describe the results you expected:**
Docker container starts successfully;

**Additional information you deem important (e.g. issue happens only occasionally):**
trying to start trying to armhf image by x86 host. QEMU-static-user has been introduced.

**Comment by thaJeztah** [comment]
Docker does not support this out of he box, so this isn't really a bug; could you try discussing this on forums.docker.com?

**Comment by kometchtech** [comment]
OK...

Try on the Forum.

**Comment by kometchtech** [comment]
Found a similar forum post seems to slightly different contents.

https://forums.docker.com/t/error-response-from-daemon-no-such-container/8618/2

So stop with the following error message in my case.
`standard_init_linux.go:175: exec user process caused "no such file or directory"`

**Comment by kometchtech** [comment]
@thaJeztah 
As the file is not found.

```
kometch@ubuntu-virt:~$ docker run -it 2a54fc0d3177 bash
standard_init_linux.go:175: exec user process caused "no such file or directory"
kometch@ubuntu-virt:~$ sudo find / -name standard_init_linux.go
[sudo] password for kometch: 
find: ‘/run/user/1000/gvfs’: Permission denied
kometch@ubuntu-virt:~$ 
```

**Comment by thaJeztah** [comment]
That file you're searching for is just one of the source files in docker; https://github.com/opencontainers/runc/blob/master/libcontainer/standard_init_linux.go

The actual "no such file" error is the Linux kernel giving a non-informative error; 

You cannot run the iot/armhf-ubuntu image, because it can only run on armhf hardware, and you are running on amd64. If you try to run such an image, it will fail and show the error you reported.

Docker for Mac *can* run some architectures, but it's an experimental feature

**Comment by kometchtech** [comment]
And you, I understand.
x 86 version, corresponding to other architectures will be?

My best regards.

**Comment by thaJeztah** [comment]
amd64 is the default; just run `docker run -it ubuntu` and it should work

**Comment by kometchtech** [comment]
However, running on the track record of the past.

http://inaz2.hatenablog.com/entry/2015/03/03/235759

That bogs down from 1.12 in this is something around the behavior was changed?

My best regards.

**Comment by thaJeztah** [comment]
I'm not aware of anything changed in that area; perhaps @justincormack has some ideas

**Comment by kfuglsang** [comment]
Im getting the same error on 1.12 (Docker for Windows)

https://forums.docker.com/t/standard-init-linux-go-175-exec-user-process-caused-no-such-file/20025

**Comment by zjxpsetp** [comment]
@thaJeztah  We got the similar error "standard_init_linux.go:175: exec user process caused "exec format error" when we try to update our docker engine from (1.7 - 1.9 ) to 1.12. And sometime we may get this error (not always), here is my docker info:

Containers: 3
 Running: 3
 Paused: 0
 Stopped: 0
Images: 3
Server Version: 1.12.1
Storage Driver: devicemapper
 Pool Name: docker-202:2-57775-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.96 GB
 Data Space Total: 107.4 GB
 Data Space Available: 20.38 GB
 Metadata Space Used: 3.83 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.144 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.77 (2012-10-15)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: null host bridge overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: apparmor
Kernel Version: 3.13.0-61-generic
Operating System: Ubuntu 14.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 7.829 GiB
Name: doctor
ID: MBX4:LN2T:RZ3I:OZYJ:T57U:KR3A:CKCJ:IDXZ:NVAN:HQOE:MDXA:TAUJ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support

**Comment by zjxpsetp** [comment]
I can start the same image in other server. And if I re-install docker engine, I can start this image too.
